### ISSUE TITLE
Anonymous labels?

### ISSUE THREAD

**Issue by neilsf** [dasm-assembler/dasm#17]
Hi. I use DASM as the intermediate compiler under XC=BASIC mainly because of its very flexible macro syntax. A feature that I miss a lot is support for anonymous labels (+, -) like in ACME for example. Are you planning to implement something similar? Thanks.

**Comment by dionoid** [comment]
Hi, DASM does support anonymous/temporary labels that begin with a dot. 
From the documentation:

	    Temporary label names are unique within segments of code bounded by SUBROUTINE:

		CHARLIE subroutine
			ldx #10
		.1	dex
			bne .1
		BEN	subroutine
			ldx #20
		.qq	dex
			bne .qq

	    Automatic temporary label boundries occur for each macro level.
	    Usually temporary labels are used in macros and within actual
	    subroutines (so you don't have to think up a thousand different
	    names)

**Comment by thrust26** [comment]
Also: Within macros you can create temporary labels using $ (e.g. name$).

**Comment by neilsf** [comment]
Yeah I'm aware of all those, but I was wondering if something like this was possible:

        ldx #20
    !   dex
        bne -

...if not, that's fair enough. Thanks for the great work.

**Comment by thrust26** [comment]
That's new to me. Might be a good new feature. 

Which assembler is supporting this?

**Comment by neilsf** [comment]
Not exactly the same syntax as in my example but here are anonymous forward/blackward labels in ACME:

https://sourceforge.net/p/acme-crossass/code-0/6/tree/trunk/docs/QuickRef.txt#l52

**Comment by neilsf** [comment]
Another example is Kick Assembler that supports this:

       ldx #10
    !loop:
       jmp !+ // Jumps over the two next nops to the ! label
       nop
    !: jmp !+ // Jumps over the two next nops to the ! label
       nop
       nop
    !:
       dex
       bne !loop- // Jumps to the previous !loop label

**Comment by thrust26** [comment]
The latter would become like this in DASM:
```   
   ldx #10
.loop
   jmp .+ // Jumps over the two next nops to the . label
   nop
   nop
.  jmp .+ // Jumps over the two next nops to the . label
   nop
   nop
.
   dex
   bne .loop- // Jumps to the previous .loop label
```
I like the idea.

**Comment by Rhialto** [comment]
A problem /might/ be there that `.` also stands for the current assembly address. I'm not sure if current code would get the right interpretation here (but then again, it might...)
